### PR TITLE
convert @babel/preset-env to typescript

### PR DESCRIPTION
**Pull request by zxbodya**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

@babel/preset-env part of #11578

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13217"></a>

### CODE REVIEW

File: packages/babel-preset-env/src/targets-parser.ts [hunk 1]
```diff
@@ -3,5 +3,4 @@
 export {
   default,
   isBrowsersQueryValid,
-  semverMin,
```
Comment (nicolo-ribaudo) [inline packages/babel-preset-env/src/targets-parser.ts:6]: This was not actually exported by `@babel/helper-compilation-targets`

**Review by nicolo-ribaudo — APPROVED** [review]

File: packages/babel-preset-env/src/debug.ts [hunk 2]
```diff
@@ -1,9 +1,6 @@
-// @flow
+import { getInclusionReasons } from "@babel/helper-compilation-targets";
 
-import {
-  getInclusionReasons,
-  type Targets,
-} from "@babel/helper-compilation-targets";
```
Comment (zxbodya) [inline packages/babel-preset-env/src/debug.ts:6]: it looks, I had eslint not working while rebasing… - this was to be fixed, by it… -- will update other typescript migration PRs

Comment (nicolo-ribaudo) [reply packages/babel-preset-env/src/debug.ts:6]: We recently updated prettier, maybe you had the old version in `node_modules`

Comment (zxbodya) [reply packages/babel-preset-env/src/debug.ts:6]: I guess, that as well… anyway rebasing, everything after reinstalling dependencies, and building the project(to have eslint working)

**Review by JLHwung — APPROVED** [review]

**Pull request merged**